README.md

```diff
@@ -233,49 +233,54 @@ Each CUDA version only supports one particular XCode version. The following comb
+It's faily easy to build with CPU. It is recommended to build the latest baseline with Visual Studio 2019 version 16.7.6 or higher. 
 
+Build with Cuda
+
+[NVTX](https://docs.nvidia.com/gameworks/content/gameworkslibrary/nvtx/nvidia_tools_extension_library_nvtx.htm) are needed to build Pytorch with Cuda
 NVTX is a part of CUDA distributive, where it is called "Nsight Compute". To install it onto already installed CUDA run CUDA installation once again and check the corresponding checkbox.
-Be sure that CUDA with Nsight Compute is installed after Visual Studio 2017.
+Be sure that CUDA with Nsight Compute is installed after Visual Studio.
+
+Currently, VS 2017 / 2019, and Ninja are supported as the generator of CMake. If `ninja.exe` is detected in `PATH`, then Ninja will be used as the default generator, otherwise, it will use VS 2017 / 2019.
+<br/> If Ninja is selected as the generator, the latest MSVC wil get selected as the underlying toolchain.
 
-Currently, VS 2017, VS 2019, and Ninja are supported as the generator of CMake. If `ninja.exe` is detected in `PATH`, then Ninja will be used as the default generator, otherwise, it will use VS 2017.
-<br/> If Ninja is selected as the generator, the latest MSVC which is newer than VS 2015 (14.0) will get selected as the underlying toolchain. If you use CMake <= 3.14.2 and has VS 2019 installed, then even if you specify VS 2017 as the generator, VS 2019 will get selected as the generator.
+CUDA and MSVC, MSVC and Pytorch have version dependency, please install the matched versions. 
+| CUDA version | Newest supported VS version                             | Pytorch version |
+| ------------ | ------------------------------------------------------- | --------------- |
+| 9.2          | Visual Studio 2017 Update 5 (15.5) (`_MSC_VER` <= 1912) |       <= 1.6    |
+| 10.0         | Visual Studio 2017 (15.X) (`_MSC_VER` < 1920)           |       <= 1.6    |
+| 10.1 ~ 11.0  | Visual Studio 2019 (16.X) (`_MSC_VER` < 1930)           |       >= 1.7    |
 
-CUDA and MSVC have strong version dependencies, so even if you use VS 2017 / 2019, you will get build errors like `nvcc fatal : Host compiler targets unsupported OS`. For this kind of problem, please install the corresponding VS toolchain in the table below, and then you can either specify the toolset during activation (recommended) or set `CUDAHOSTCXX` to override the Cuda host compiler (not recommended if there are big version differences).
+[Magma](https://developer.nvidia.com/magma), [oneDNN, a.k.a MKLDNN or DNNL](https://github.com/oneapi-src/oneDNN), [Sccache](https://github.com/oneapi-src/oneDNN) are often needed. Please refer  https://github.com/pytorch/pytorch/tree/master/.jenkins/pytorch/win-test-helpers/installation-helpers to install them.
+
+Some environment variables could be refer https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/win-test-helpers/build_pytorch.bat
```

Comment:
```suggestion
You can refer to [build_pytorch.bat](https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/win-test-helpers/build_pytorch.bat) script for environment variable configurations.
```
Also I am not sure what this paragraph has to do with the previous one. (might be in the wrong place?)

Reply:
Does the previous one refer to those additional libraries, Magma, oneDNN and sccache?
Not like Linux, besides those additional libs, users have to set CUDA_PATH, CUDNN_LIB_DIR, and other envs in windows.
If they uses Sccache, they have to set some [special envs](https://github.com/pytorch/pytorch/blob/8ff0b6fef822edae1b557649b09dccfdaa52a2e3/.jenkins/pytorch/win-test-helpers/build_pytorch.bat#L94-L110) to avoid intermittent [build error](https://github.com/pytorch/pytorch/issues/25393).